core/src/test/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilderTest.java

```diff
@@ -0,0 +1,222 @@
+import static org.elasticsearch.index.query.GeoBoundingBoxQueryBuilder.MAX_LAT;
+import static org.elasticsearch.index.query.GeoBoundingBoxQueryBuilder.MIN_LON;
+import static org.elasticsearch.index.query.GeoBoundingBoxQueryBuilder.MAX_LON;
+
+@SuppressWarnings("deprecation")
+public class GeoBoundingBoxQueryBuilderTest extends BaseQueryTestCase<GeoBoundingBoxQueryBuilder> {
+    @Override
+    protected GeoBoundingBoxQueryBuilder doCreateTestQueryBuilder() {
+        GeoBoundingBoxQueryBuilder builder = new GeoBoundingBoxQueryBuilder("mapped_geo");
+
+        if (randomBoolean()) {
+            builder.queryName(randomAsciiOfLengthBetween(1, 10));
+        }
+
+        double top = randomDoubleBetween(MIN_LAT, MAX_LAT);
+        double bottom = randomDoubleBetween(top, MAX_LAT);
+        double left = randomDoubleBetween(MIN_LON, MAX_LON);
+        double right = randomDoubleBetween(left, MAX_LON);
+
+        if (frequently()) {
+            // check the top-left/bottom-right combination of setters
+            int path = randomIntBetween(0, 2);
+            switch(path) {
+                case 0: 
+                    builder.topLeft(new GeoPoint(top, left));
+                    break;
+                case 1:
+                    builder.topLeft(GeohashUtils.encodeLatLon(top, left));
+                    break;
+                default:
+                    builder.topLeft(top, left);
+            }
+
+            path = randomIntBetween(0, 2);
+            switch(path) {
+            case 0: 
+                builder.bottomRight(new GeoPoint(bottom, right));
+                break;
+            case 1:
+                builder.bottomRight(GeohashUtils.encodeLatLon(bottom, right));
+            default:
+                builder.bottomRight(bottom, right);
+            }
+        } else {
+            // check the deprecated top-right/bottom-left combination of setters
+            int path = randomIntBetween(0, 2);
+            switch(path) {
+                case 0: 
+                    builder.topRight(new GeoPoint(top, right));
+                    break;
+                case 1:
+                    builder.topRight(GeohashUtils.encodeLatLon(top, right));
+                    break;
+                default:
+                    builder.topRight(top, right);
+            }
+
+            path = randomIntBetween(0, 2);
+            switch(path) {
+            case 0: 
+                builder.bottomLeft(new GeoPoint(bottom, left));
+                break;
+            case 1:
+                builder.bottomLeft(GeohashUtils.encodeLatLon(bottom, left));
+            default:
+                builder.bottomLeft(bottom, left);
+            }
+        }
+
+        // TODO we aren't testing the "memory" scenario - needs a node pulled up for testing
+        builder.type(Type.INDEXED);
+        return builder;
+    }
+
+    private double randomDoubleBetween(double start, double end) {
+        return start + (randomDouble() * (end - start));
```

Comment:
Minor nitpick: according to the docs, Random.nextDouble (randomDouble() uses this) returns [0.0, 1.0), so this would return [start, end) I think (see usage below).